core/server/worker/src/main/java/alluxio/worker/block/TieredBlockStore.java

```diff
@@ -676,7 +676,7 @@ private TempBlockMeta createBlockMetaInternal(long sessionId, long blockId,
   private void freeSpaceInternal(long sessionId, long availableBytes, BlockStoreLocation location)
       throws WorkerOutOfSpaceException, IOException {
     EvictionPlan plan;
-    try (LockResource r = new LockResource(mMetadataReadLock)) {
+    try (LockResource r = new LockResource(mMetadataWriteLock)) {
```

Comment:
can you add a comment here
mentioning that grabbing the write lock due to ALLUXIO-3089?
Ideally this should still use read lock